File: ShoppingAssistant.Api/ShoppingAssistant.Api/Models/Store.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace ShoppingAssistant.Api.Models
{
    public class Store
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Location { get; set; }
    }
}

File: ShoppingAssistant.Mobile/shopping_assistant/lib/models/store.dart
class Store {
  String id;
  String name;
  String address;
  String location;

  Store({
    required this.id,
    required this.name,
    required this.address,
    required this.location,
  });

  Store.fromJson(Map<String, dynamic> json)
      : id = json['id'] as String,
        name = json['name'] as String,
        address = json['address'] as String,
        location = json['location'] as String;

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'address': address,
      'location': location,
    };
  }
}
